pkg/pillar/cmd/volumemgr/create.go

```diff
@@ -126,7 +154,7 @@ func createContainerVolume(ctx *volumemgrContext, status types.VolumeStatus,
 func destroyVolume(ctx *volumemgrContext, status types.VolumeStatus) (bool, string, error) {
 
 	log.Functionf("destroyVolume(%s)", status.Key())
-	if !status.VolumeCreated {
+	if status.SubState != types.VolumeSubStateCreated {
```

Comment:
Is there something we need to clean up if we did the prepare and not the create?

Reply:
Good question. Seems, we need to remove zvol in this case, it is creating on preparation step.

Reply:
Seems, that in this case (zfs) un-prepare equals to clean, so, seems, I should check here Prepared or Created for zfs